easybuild/tools/include.py

```diff
@@ -0,0 +1,210 @@
+
+@author: Kenneth Hoste (Ghent University)
+"""
+import glob
+import os
+import sys
+from vsc.utils.missing import nub
+from vsc.utils import fancylogger
+
+from easybuild.tools.build_log import EasyBuildError
+# these are imported just to we can reload them later
+import easybuild.tools.module_naming_scheme
+import easybuild.toolchains
+import easybuild.toolchains.compiler
+import easybuild.toolchains.fft
+import easybuild.toolchains.linalg
+import easybuild.toolchains.mpi
+# importing easyblocks namespace may fail if easybuild-easyblocks is not available
+# for now, we don't really care
+try:
+    import easybuild.easyblocks
+    import easybuild.easyblocks.generic
+except ImportError:
+    pass
+
+
+_log = fancylogger.getLogger('tools.include', fname=False)
+
+
+PKG_INIT_BODY = """
+from pkgutil import extend_path
+__path__ = extend_path(__path__, __name__)
+"""
+
+def create_pkg(path):
+    """Write package __init__.py file at specified path."""
+    init_path = os.path.join(path, '__init__.py')
+    try:
+        # note: can't use mkdir, since that required build options to be initialised
+        if not os.path.exists(path):
+            os.makedirs(path)
+
+        # put __init__.py files in place, with required pkgutil.extend_path statement
+        # note: can't use write_file, since that required build options to be initialised
+        handle = open(init_path, 'w')
+        handle.write(PKG_INIT_BODY)
+        handle.close()
+    except (IOError, OSError) as err:
+        raise EasyBuildError("Failed to write %s: %s", init_path, err)
+
+
+def set_up_eb_package(parent_path, eb_pkg_name, subpkgs=None):
+    """Set up new easybuild subnamespace in specified path."""
+    if not eb_pkg_name.startswith('easybuild'):
+        raise EasyBuildError("Specified EasyBuild package name does not start with 'easybuild': %s", eb_pkg_name)
+
+    pkgpath = os.path.join(parent_path, eb_pkg_name.replace('.', os.path.sep))
+
+    # handle subpackages first
+    if subpkgs:
+        for subpkg in subpkgs:
+            create_pkg(os.path.join(pkgpath, subpkg))
+
+    # creata package dirs on each level
+    while pkgpath != parent_path:
+        create_pkg(pkgpath)
+        pkgpath = os.path.dirname(pkgpath)
+
+
+def expand_glob_paths(glob_paths):
+    """Expand specified glob paths to a list of unique non-glob paths to only files."""
+    paths = []
+    for glob_path in glob_paths:
+        paths.extend([f for f in glob.glob(glob_path) if os.path.isfile(f)])
+
+    return nub(paths)
+
+
+def safe_symlink(source_path, symlink_path):
+    """Create a symlink at the specified path for the given path."""
+    try:
+        os.symlink(os.path.abspath(source_path), symlink_path)
+        _log.info("Symlinked %s to %s", source_path, symlink_path)
+    except OSError as err:
+        raise EasyBuildError("Symlinking %s to %s failed: %s", source_path, symlink_path, err)
```

Comment:
I don't want to get a traceback for something like this, just a clear error suffices. Tracebacks confuse regular users, and have no extra value in this case (symlinking would only fail here in case of some system error).